plugins/node/opentelemetry-instrumentation-restify/src/instrumentation.ts

```diff
@@ -0,0 +1,204 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import * as api from '@opentelemetry/api';
+import * as restify from 'restify';
+import { Server } from 'restify';
+import * as types from './types';
+import { VERSION } from './version';
+import once = require('lodash.once');
+import {
+  InstrumentationBase,
+  InstrumentationNodeModuleDefinition,
+  InstrumentationNodeModuleFile,
+  isWrapped,
+} from '@opentelemetry/instrumentation';
+import { HttpAttribute } from '@opentelemetry/semantic-conventions';
+
+const { diag } = api;
+const RESTIFY_MW_METHODS = ['use', 'pre'];
+const RESTIFY_METHODS = ['del', 'get', 'head', 'opts', 'post', 'put', 'patch'];
```

Comment:
Does restify support custom HTTP methods? If so, will this instrumentation be able to work with them?

Reply:
No. Everything that allows adding a handler is listed here. There's a "wildcard"-way of reaching to the API of [find-my-way](https://github.com/delvedor/find-my-way) which is mentioned [here](http://restify.com/docs/server-api/#servermethodopts) in `restify` docs, but even that has to be done through one of these functions.

However, I'm not testing for that shortcut. Will add a test for that.

Reply:
Restify does not support custom HTTP methods, it turns out. It overwrites it with the one specified with function used to register the handler. Added a test to make sure the more verbose API works as expected.